src/devices/SoftwareSpi/SoftwareSpi.cs

```diff
@@ -145,7 +145,7 @@ public override void TransferFullDuplex(ReadOnlySpan<byte> dataToWrite, Span<byt
         {
             if (dataToWrite.Length != dataToRead.Length)
             {
-                throw new ArgumentException(nameof(dataToWrite));
+                throw new ArgumentException(null, nameof(dataToWrite));
```

Comment:
I'm not sure passing null here is a good idea. Wouldn't this possibly lead to a NullReferenceException when handling the error?

Reply:
This is what visual studio code fix does by default. I could replace null with actual message describing what caused the exception. 

![ca2208-codefix_null_msg](https://user-images.githubusercontent.com/34100964/162391932-ec4e51dd-d04c-4abc-bd65-d154f85195a9.png)

Reply:
That would be the ideal case, for sure.

Reply:
Replaced null parameters with error messages.